### PR TITLE
Added Sitemap plugin and config

### PR DESCRIPTION
**Pull request by mehzer**
Addresses #466 by using https://github.com/jekyll/jekyll-sitemap 
Adjusted _config to add url: and the plugin reference

Produces pulumi.io/sitemap.xml

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -8,6 +8,9 @@
 # jekyll-redirect-from, and we don't need those to be fully qualified.
 pulumi_url: https://pulumi.io
 
+# Needed for jekyll-sitemap
+url: "https://pulumi.io"
```
Comment (justinvp) [inline _config.yml]: Now that we need this for the sitemap, would you mind adding a commit that removes the `pulumi_url` variable above along with the comment, and replace it's use in `head.html` (e.g. `site.pulumi_url` => `site.url`)?

File: _config.yml [hunk 2]
```diff
@@ -18,6 +21,7 @@ plugins:
   - jekyll-redirect-from # for 301 redirects
   - jekyll-relative-links # to convert internal MD links to HTML
   - jekyll-optional-front-matter # for Markdown without front matter (e.g., CLI reference)
+  - jekyll-sitemap # produce a sitemap.xml file for submission to search engines
```
Comment (justinvp) [inline _config.yml]: We should exclude `sitemap.xml` from the search index by adding a check to `search-data.json`.

File: _config.yml (hunk 1 above)
Comment (mehzer) [reply _config.yml]: Made this change

File: _config.yml (hunk 2 above)
Comment (mehzer) [reply _config.yml]: Made this change

**Comment by mehzer** [comment]
Fixed per your feedback. Not sure on other exclusions at this time.

**Pull request merged**